src/main/java/com/mongodb/spark/sql/connector/schema/BsonDocumentToRowConverter.java

```diff
@@ -212,7 +243,7 @@ private GenericRowWithSchema convertToRow(
         .forEach((k, v) ->
             map.put(k, convertBsonValue(createFieldPath(fieldName, k), dataType.valueType(), v)));
 
-    return JavaScala.asScala(map);
+    return scala.collection.immutable.Map$.MODULE$.from(JavaScala.asScala(map));
```

Comment:
Found this bug when converting a row to json and Spark errored due to it not being the correct (immutable) scala map type.